### PR TITLE
Support `Modify Column` for MySQL dialect

### PR DESCRIPTION
**Pull request by KKould**
``` sql
ALTER TABLE t1 MODIFY b INT
```
(see https://dev.mysql.com/doc/refman/8.0/en/alter-table.html).

### CODE REVIEW

**Review by iffyio — COMMENTED** [review]
Hi @KKould! could you add tests to [mysql](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/tests/sqlparser_mysql.rs#L2002) demonstrating the behavior?

**Comment by KKould** [comment]
> Hi @KKould! could you add tests to [mysql](https://github.com/sqlparser-rs/sqlparser-rs/blob/main/tests/sqlparser_mysql.rs#L2002) demonstrating the behavior?

thank you for the reminder, please check again

**Review by iffyio — COMMENTED** [review]
Thanks @KKould! Left a comment re the column keyword, other than that the changes look reasonable to me

File: src/ast/ddl.rs [hunk 1]
```diff
@@ -292,6 +300,22 @@ impl fmt::Display for AlterTableOperation {
 
                 Ok(())
             }
+            AlterTableOperation::ModifyColumn {
+                col_name,
+                data_type,
+                options,
+                column_position,
+            } => {
+                write!(f, "MODIFY COLUMN {col_name} {data_type}")?;
```
[the remark this answers is not in the document]
Comment (iffyio) [reply src/ast/ddl.rs:309]: oh I see, yeah no need to tackle that one in this PR unless you feel its straight-forward and want to. for this PR main would be to ensure the behavior for modify column

[the remark this answers is not in the document]
Comment (iffyio) [reply src/ast/ddl.rs:309]: @KKould double checking: did you plan to update the PR for the `MODIFY COLUMN` variant introduced?

**Comment by iffyio** [comment]
cc @alamb changes LGTM for review

File: src/ast/ddl.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (alamb) [reply src/ast/ddl.rs:309]: I agree it would be great to ensure this type of syntax roundtrips but no need to do it in this PR

**Review by alamb — APPROVED** [review]

**Pull request merged**